core/core-spacemacs.el

```diff
@@ -82,6 +82,9 @@ the final step of executing code in `emacs-startup-hook'.")
   (setq dotspacemacs-editing-style (dotspacemacs//read-editing-style-config
                                     dotspacemacs-editing-style))
   (configuration-layer/initialize)
+  ;; frame title init
+  (when (and (display-graphic-p) dotspacemacs-frame-title-format)
+    (setq frame-title-format '((:eval (spacemacs/frame-title-prepare)))))
```

Comment:
From documentation (GNU Emacs 24.5.1):
> Template for displaying the title bar of an iconified frame.
> (Assuming the window manager supports this feature.)

You can see iconified frames with title bars in a tiling windows manager
with a tabs support. When one window is displaying and other are not,
that windows that are not displaying are called "tabbified" or
"iconfied". But their title bars are still displaying as a tabs.

So without setting this variable you will get different frame name
format for different frames.

Despite the fact that this behavior irritates me, I would say that for
some use-cases it can be useful.